src/org/thoughtcrime/securesms/mms/AttachmentManager.java

```diff
@@ -424,8 +424,11 @@ public static void selectLocation(Activity activity, int chatId) {
       return;
     }
 
+    // see https://support.google.com/googleplay/android-developer/answer/9799150#zippy=%2Cstep-provide-prominent-in-app-disclosure
+    // for rationale dialog requirements
     Permissions.PermissionsBuilder permissionsBuilder = Permissions.with(activity)
             .ifNecessary()
+            .withRationaleDialog("This app will collect location data to enable sharing your locations with chat members for a given time frame.\n\nLocation data is collected even when the app is closed or not in use.", R.drawable.ic_location_on_white_24dp)
```

Comment:
I don't know if this is the time to discuss the wording, but I'll just go ahead, if not just merge it without changing it:
- I'd drop out "for a given time frame" as it will be clear to users that the location streaming can be stopped again
- I'd use "location streaming" or "live location"
- Why "collect"?
- At https://support.google.com/googleplay/android-developer/answer/9799150?hl=en#zippy=%2Cstep-provide-prominent-in-app-disclosure%2Csee-an-example-of-prominent-in-app-disclosure it says we shall explain _why_ we want location access while DC is in the background.

All in all, this would give sth like:

```
To share your live location with your friends, allow Delta Chat to access your location.

If you want to continue location streaming while the app is in the background, allow access all of the time.
```

Reply:
the fist part is better, i will change that. however, "friends" is a bit unspecific, we know it more concrete at that moment, things are shared with the "chat members", so maybe better stay with that.

the second part, "If you want to continue location streaming while the app is in the background, ..." does not work if the user does not have a choice.

Reply:
@Hocuri maybe that one:

```
To share your live location with chat members,
allow Delta Chat to use your location data.

Background location is used
to make live location work gaplessly
even when the app is closed or not in use.
```

Reply:
> does not work if the user does not have a choice.

I thought they do have a choice? They can choose "Allow all the time", "Allow only while using the app" and "Deny" when the system asks.

Anyway, your version is nice too.